File: solutions/d1000000/Kotlin/d1000000.kt
import java.io.BufferedReader
import java.io.InputStreamReader

fun maxStraightLength(list: List<Int>): Int {
    var maxLength = 0
    val sortedList = list.sorted()

    for (j in sortedList.indices) {
        if (sortedList[j] > maxLength) {
            maxLength++
        }
    }
    return maxLength
}

fun main() {
    val reader = BufferedReader(InputStreamReader(System.`in`))
    val testCases = reader.readLine().toInt()

    for (i in 1..testCases) {
        reader.readLine() // Skip the number of dice input as it's never used
        val diceIn = reader.readLine()

        val dice = diceIn.split(" ").map { it.toInt() }
        val res = maxStraightLength(dice)
        println("Case #$i: $res")
    }
}

File: solutions/d1000000/Dart/d1000000.dart
// this is the suggested approach in the analysis page
// as is this code gives the correct answer but exceeds the time limit
// last updated Wed 19 Apr 15:00

// dart:io needs to be imported to read inputs
import 'dart:io';

// this function starts at 1 and counts to the highest number possible
// the highest number possible is also the length of the longest straight
int maxStraightLength(list) {
  var maxLength = 0;
  list.sort();

  for (int j = 0; j < list.length; j++){
    if (list[j] > maxLength){
      maxLength++;
    }
  }
  return maxLength;
}

void main() {
  // reading number of tests from input
  var testS = stdin.readLineSync();
  int testCases = testS != null ? int.parse(testS) : 0;

  for (int i = 0; i < testCases; i++) {
    // reading number of dice and their value
    var nDiceS = stdin.readLineSync();
    int nDice = nDiceS != null ? int.parse(nDiceS) : 0;
    var diceIn = stdin.readLineSync();

    // converting string to list of strings
    List<String> diceS = diceIn != null ? diceIn.split(' ') : [];
    // converting list of strings to list of numbers
    List<int> dice = diceS.map(int.parse).toList();

    int res = maxStraightLength(dice);
    print("Case #${i + 1}: $res");
  }
}